### PR TITLE
Per-instance based configuration

### PR DESCRIPTION
**Pull request by richardtop**
Fix for #197

Now it is possible to configure each ImagePicker individually, on per-instance basis.
To have a custom configuration, ImagePickerController takes it as an initialization parameter and propagates further to the dependable classes. If the configuration is `nil`, it uses default one.

Changes in short:
1. Made static variables instance in `Configuration`
2. To each class that depends on `Configuration` added it as an instance variable
3. Added `Configuration` as an optional parameter to all dependable classes
4. Made all the classes use instance variable configuration instead of the global one

### CODE REVIEW

**Comment by onmyway133** [comment]
This is great

File: Source/BottomView/BottomContainerView.swift [hunk 1]
```diff
@@ -62,29 +64,35 @@ open class BottomContainerView: UIView {
 
   // MARK: Initializers
 
-  public override init(frame: CGRect) {
-    super.init(frame: frame)
+  public init(configuration: Configuration?) {
```
Comment (onmyway133) [inline Source/BottomView/BottomContainerView.swift]: I personally prefer this to be non optional, using just default parameter of `Configuration()`

Comment (richardtop) [inline Source/BottomView/BottomContainerView.swift]: @onmyway133 what if other initializer is called, e.g. `init(frame: )`? Then, the configuration will not be initialized. So, I've decided to go for optional. 
How about setting default to `nil` ?

**Comment by zenangst** [comment]
This is nice, but this looks like a breaking change as `Configuration` no longer has static variables. Not really an issue but we need to bump the major version number when releasing a new version.

File: Source/Configuration.swift [hunk 2]
```diff
@@ -4,53 +4,55 @@ public struct Configuration {
...
 
   // MARK: Fonts
 
-  public static var numberLabelFont = UIFont(name: "HelveticaNeue-Bold", size: 19)!
-  public static var doneButton = UIFont(name: "HelveticaNeue-Medium", size: 19)!
-  public static var flashButton = UIFont(name: "HelveticaNeue-Medium", size: 12)!
-  public static var noImagesFont = UIFont(name: "HelveticaNeue-Medium", size: 18)!
-  public static var noCameraFont = UIFont(name: "HelveticaNeue-Medium", size: 18)!
-  public static var settingsFont = UIFont(name: "HelveticaNeue-Medium", size: 16)!
+  public var numberLabelFont = UIFont(name: "HelveticaNeue-Bold", size: 19)!
```
Comment (onmyway133) [inline Source/Configuration.swift]: @richardtop care to also use system fonts, as you described here https://github.com/hyperoslo/ImagePicker/issues/256 😉

Comment (onmyway133) [inline Source/Configuration.swift]: On a 2nd thought, I think that is outside the scope of this PR

**Comment by onmyway133** [comment]
@richardtop you rock 🚀

File: Source/Configuration.swift (hunk 2 above)
Comment (richardtop) [inline Source/Configuration.swift]: @onmyway133 exactly.
